pkg/util/arm/arm.go

```diff
@@ -11,26 +11,45 @@ import (
 	mgmtfeatures "github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-07-01/features"
 	"github.com/Azure/go-autorest/autorest"
 	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/sirupsen/logrus"
 
 	"github.com/Azure/ARO-RP/pkg/api"
-	"github.com/Azure/ARO-RP/pkg/util/arm"
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/features"
 	"github.com/Azure/ARO-RP/pkg/util/azureerrors"
 )
 
-func (m *manager) deployARMTemplate(ctx context.Context, rg string, tName string, t *arm.Template, params map[string]interface{}) error {
-	m.log.Printf("deploying %s template", tName)
+type Manager interface {
+	Deploy(ctx context.Context, resourceGroup string, deploymentName string, template *Template, parameters map[string]interface{}) error
+}
+
+type manager struct {
+	log         *logrus.Entry
+	deployments features.DeploymentsClient
+}
+
+func NewManager(azEnv *azure.Environment, log *logrus.Entry, subscriptionID string, authorizer autorest.Authorizer) Manager {
```

Comment:
I'm not convinced that moving this out of pkg/cluster is a good idea.
1) the wrapping logic is pretty specific to pkg/cluster.  I'm not sure I'd want other people to reuse this.
2) making this mockable is borderline necessary - mock_deployments pretty much already does the job for you

Reply:
Its one of those where 2 weeks ago when I wrote this code it was needed. Now I dont remember why due to number of multitasking :/ and here it goes again...